### PR TITLE
Remove location property in worklets for production builds

### PR DESCRIPTION
**Pull request by tjzel**
## Summary

Currently we inject `location` property in worklets - which is there for debugging purposes - even in production build. This PR corrects this.

Fixes: #4689

## Test plan

Added a new test suite in plugin. Also ran production android Example just to be sure.

### CODE REVIEW

File: __tests__/__snapshots__/plugin.test.ts.snap [hunk 1]
```diff
@@ -53,6 +71,21 @@ var f = function () {
 }();"
 `;
 
+exports[`babel plugin doesn't inject location for worklets in production builds 1`] = `
+"var _worklet_1852213185147_init_data = {
+  code: "function anonymous(){const x=1;}"
```
Comment (tomekzaw) [inline __tests__/__snapshots__/plugin.test.ts.snap:76]: I'm quite worried about the removal of `location` property, looks like we still use it here:

https://github.com/software-mansion/react-native-reanimated/blob/c6e7523541567a38dc4cb8d9c705b11e746a8cdf/src/reanimated2/initializers.ts#L47-L50

Maybe we should use `global.eval` when not in `__DEV__` mode?

Comment (piaskowyk) [reply __tests__/__snapshots__/plugin.test.ts.snap:76]: I agree with Tomek. 👍
Moreover, maybe we can use `location` and `sourceMap` only in debug mode. I could have a positive performance impact.

Comment (tjzel) [reply __tests__/__snapshots__/plugin.test.ts.snap:76]: We use it in debug only. `evalWithSourceMap` is injected only for dev builds.

**Review by tomekzaw — APPROVED** [review]
Thanks for optimizing this! 👍

**Review by piaskowyk — APPROVED** [review]

**Pull request merged**